elyra/pipeline/processor_kfp.py

```diff
@@ -116,6 +118,18 @@ def process(self, pipeline):
             else:
                 raise lve
 
+        try:
+            client.list_experiments(namespace=user_namespace,
+                                    page_size=1)
+        except ApiException as ae:
+            self.log.error(f'Could not create experiment {experiment_name}: {ae.reason} ({ae.status})')
+            if ae.body:
+                error_msg = json.loads(ae.body)
+                raise RuntimeError(f'Could not create experiment {experiment_name}: {ae.reason} ({ae.status}): ' +
```

Comment:
is the idea here that "list experiments" will fail to retrieve experiments if the namespace is invalid? In this case, we should raise an error or provide a more specific error message to the user (e.g. Invalid namespace). I think it's a little misleading to say we could not create an experiment before we actually submit it.

Also, are we using experiment on any other place? Otherwise I believe we should use 'jobs' which is more generic.

Reply:
Yeah I was pondering options on this yesterday. I kept this experiment wording from further down in the file (lines 193-194) where the error came directly from the `create_experiment` function. 

I agree that it's misleading now that we've changed the trapping mechanism, but I hesitated to change it to something like "invalid namespace" because `list_experiments` could technically fail for other reasons since the exception is originating as a result of a GET request. I haven't encountered any other scenarios, so maybe I'm overthinking it. 

Maybe something generic like "Error making request to pipeline server"? Or should I just assume namespace-only errors and go with a specific message like "Invalid namespace in runtime configuration"

Reply:
Neither, because the messages should express what we were trying to do: "Error validating user namespace XXX ...". The fact that we are using an experiment related API call to perform the validation is not relevant to the user. We just had to pick one that utilizes the namespace information if it is provided.

Reply:
That is super helpful to my understanding of this, thank you! That's exactly what I was trying to say but I couldn't quite get there -- accurate error messages are an art